packages/fluxible/utils/callAction.js

```diff
@@ -36,14 +37,7 @@ function callAction (actionContext, action, payload, done) {
     });
 
     if (done) {
-        executeActionPromise
-            .then(function(result) {
-                // Ensures that errors in callback are not swallowed by promise
-                setImmediate(done, null, result);
-            }, function (err) {
-                // Ensures that errors in callback are not swallowed by promise
-                setImmediate(done, err);
-            });
+        promiseCallback(executeActionPromise, done, {optimize: actionContext.optimizePromiseCallback});
```

Comment:
Extracting this to a util function so that other places (where promises need to call a callback) can use the same code.